components/scream/src/share/grid/remap/horizontal_remap_utility.hpp

```diff
@@ -0,0 +1,564 @@
+      for (int kk=0; kk<src_data_sub.extent(0); kk++) {
+        ret_h(kk) = ret_h(kk) + src_data_sub(kk)*weights_h(ii);
+      }
+    }
+    Kokkos::deep_copy(ret,ret_h);
+    return ret;
+  }
+/*---------------------------------------------*/
+  template <typename ScalarT>
+  ScalarT apply_segment(
+      const view_1d<ScalarT>& source_data)
+  {
+    ScalarT ret = 0.0;
+    auto source_data_h = Kokkos::create_mirror_view(source_data);
+    auto source_idx_h  = Kokkos::create_mirror_view(source_idx);
+    auto weights_h     = Kokkos::create_mirror_view(weights);
+    Kokkos::deep_copy(source_data_h,source_data);
+    Kokkos::deep_copy(source_idx_h,source_idx);
+    Kokkos::deep_copy(weights_h   ,weights);
+    for (int ii=0; ii< m_length; ii++) {
+      int idx = source_idx_h(ii);
+      ret = ret + source_data_h(idx)*weights_h(ii);
+    }
+    return ret;
+  }
+/*---------------------------------------------*/
+  // Useful for debugging, print to screen segment
+  // information
+  void print_seg()
+  {
+    auto source_dofs_h = Kokkos::create_mirror_view(source_dofs);
+    auto source_idx_h  = Kokkos::create_mirror_view(source_idx);
+    auto weights_h     = Kokkos::create_mirror_view(weights);
+    Kokkos::deep_copy(source_dofs_h,source_dofs);
+    Kokkos::deep_copy(source_idx_h ,source_idx );
+    Kokkos::deep_copy(weights_h    ,weights    );
+    printf("\n--------------------\n");
+    printf("Printing information for segment with DOF = %d\n",m_dof);
+    printf("  length = %d\n",m_length);
+    printf("%10s: %10s, %10s, %s\n","ii","source dof","source idx","weight");
+    for (int ii=0; ii<m_length; ii++) {
+      printf("%10d: %10d, %10d, %e\n",ii, source_dofs_h(ii), source_idx_h(ii), weights_h(ii));
+    }
+    printf("\n--------------------\n");
+  }
+/*---------------------------------------------*/
+
+}; // end struct RemapSegment
+
+/* --------------------------------------------------------------------------------------------- */
+/* A structure to store all of the map information 
+ * --------------------------------------------------------------------------------------------- */
+struct GSMap { // TODO: Following the name in component coupler, could be a different name
+
+using gid_type = AbstractGrid::gid_type;
+using Pack = ekat::Pack<Real,SCREAM_PACK_SIZE>;
+
+using KT = KokkosTypes<DefaultDevice>;
+
+template <typename S>
+using view_1d = typename KT::template view_1d<S>;
+
+template <typename S>
+using view_2d = typename KT::template view_2d<S>;
+
+template <typename S>
+using view_3d = typename KT::template view_3d<S>;
+
+template <typename S>
+using view_1d_host = typename KT::template view_1d<S>::HostMirror;
+
+template <typename S>
+using view_2d_host = typename KT::template view_2d<S>::HostMirror;
+
+public:
+  virtual ~GSMap() = default;
+  GSMap() {};
+
+/*---------------------------------------------*/
+  void set_name(const std::string& name) { map_name = name; }
```

Comment:
Just curious--what's the significance of this name?

Reply:
Useful for error messages when you may have more than one horizontal remapper defined